lib/ModuleWarning.js

```diff
@@ -2,15 +2,20 @@
 	MIT License http://www.opensource.org/licenses/mit-license.php
 	Author Tobias Koppers @sokra
 */
-function ModuleWarning(module, warning) {
-	Error.call(this);
-	Error.captureStackTrace(this, ModuleWarning);
-	this.name = "ModuleWarning";
-	this.module = module;
-	this.message = warning;
-	this.warning = warning;
+"use strict";
+
+class ModuleWarning extends Error {
+	constructor(module, warning) {
+		super();
+
+		if(Error.hasOwnProperty("captureStackTrace")) {
+			Error.captureStackTrace(this, ModuleWarning);
```

Comment:
Not needed in native subclasses, unless you specifically want to exclude the constructor from the stack trace.

```
CustomErrorClass: without captureStackTrace
    at CustomErrorClass (.../test.js:5:10)
    at throwMe (.../test.js:32:11)
CustomErrorClass2: with captureStackTrace
    at throwMe (.../test.js:32:11)
```

ES5-style errors would just have no `.stack` at all, so it was necessary.